Sources/TuistAsyncQueue/AsyncQueue.swift

```diff
@@ -108,7 +111,8 @@ public class AsyncQueue: AsyncQueuing {
     private func dispatchPersisted(eventTuple: AsyncQueueEventTuple) async throws {
         guard let dispatcher = dispatchers.first(where: { $0.key == eventTuple.dispatcherId })?.value else {
             try await deletePersistedEvent(filename: eventTuple.filename)
-            logger.error("Couldn't find dispatcher for persisted event with id: \(eventTuple.dispatcherId)")
+            ServiceContext.current?.logger?
+                .error("Couldn't find dispatcher for persisted event with id: \(eventTuple.dispatcherId)")
```

Comment:
Should we consider creating a helped method to make the API a bit more convenient?

Such as:
```swift
// TuistSupport/logger.swift

public func logger() -> Logging {
  ServiceContext.current?.logger // Not sure if it would make sense to provide a default value if the logger is not present
}

// Usage
logger().error(...) // Instead of ServiceContext.current?.logger?.error(...)
```

I'm generally fine with the current API as well, but given we use logger really often, it might be worth adding some syntax sugar.

Reply:
I thought about that, but I ended up not doing it because it'd tie the logger to how it's being injected/used, which I don't think the logger should be concerned with.
Alternatively, I thought about exposing it in `ServiceContext`, but we'd go from: `ServiceContext.current?.logger` to `ServiceContext.currentLogger` so I don't think it was worth it. I think the current approach is very explicit about what's going on.

Reply:
I tend to agree with @fortmarek, I would avoid this verbosity for something as common as a logger.

Reply:
Maybe some static helper so you can use `Logger.error(…)` directly?

Reply:
I disagree. A logger should not be concerned about its usage. It's a detail of the caller. In any case, the service context could expose a convenient method ServiceContext.logger instead of ServiceContext.current.logger, but I don't think we are gaining much there, so I'd rather keep things explicit instead of hiding the detail of how dependencies are being injected.